File: lc1108_defangip.c
char * defangIPaddr(char * address){
    char* defang;
    defang=malloc(strlen(address)+1+6);
    int i=0;
    int j=0;
    for(i=0; i<strlen(address); i++){
        if(address[i] == '.'){
            defang[j++]='[';
            defang[j++]='.';
            defang[j++]=']';
        }else{
            defang[j++]=address[i];
        }
    }
    defang[j]='\0';
    return defang;
}

File: lc1108_defangip.py
class Solution:
    def defangIPaddr(self, address: str) -> str:
        i=0
        j=0
        strlist=list(address)
        defang=[]
        while i< len(strlist):
            if strlist[i] == '.':
                 defang.append('[')
                 defang.append('.')
                 defang.append(']')
            else:     
                defang.append(address[i])
            i+=1
        retstr=""
        
    
        # return string   
        return (retstr.join(defang))
        
            
